AllReadyApp/Web-App/AllReady/Features/Notifications/VolunteerAssignedToItineraryHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System.Threading.Tasks;
+using AllReady.Models;
+using MediatR;
+using Microsoft.Data.Entity;
+
+namespace AllReady.Features.Notifications
+{
+    public class VolunteerAssignedToItineraryHandler : IAsyncNotificationHandler<VolunteerAssignedToItinerary>
+    {
+        private readonly AllReadyContext _context;
+        private readonly IMediator _mediator;
+
+        public VolunteerAssignedToItineraryHandler(AllReadyContext context, IMediator mediator)
+        {
+            _context = context;
+            _mediator = mediator;
+        }
+
+        public async Task Handle(VolunteerAssignedToItinerary notification)
+        {
+            var taskSignup = await _context.TaskSignups
+                .AsNoTracking()
+                .Include(x => x.User)
+                .SingleAsync(ts => ts.Id == notification.TaskSignupId)
+                .ConfigureAwait(false);
```

Comment:
@dpaquette, it is my understanding when you don't need the original calling context back after invoking an await call, you should explicitly set ConfigureAwait(false) for best performance.

An example of where you would not want to use ConfigureAwait(false) is in a UI thread, that would need to pick it's context back up again after the awaited call is finished to it can return a result to the UI. 

For example. I would not want to use .ConfigureAwait(false) in an MVC controller action.

Sean Feldman gave me some guidance early on in this project with using .ConfigureAwait(false) in handler code. You can see some of his feedback here on one of my PR's. He corrects me to use .ConfigureAwait(false) both on async invocations of the mediator and db context:
https://github.com/HTBox/allReady/pull/638#issuecomment-204093377

That being said:
1. I might not have interpreted Sean's guidance correctly.
2. with .NET Core, there might be a better/different way to do this
3. I don't understand the intricacies of ConfigureAwait yet, and I should not be using it until I understand what I'm doing ;)